integration-test/src/test/java/com/datastrato/gravitino/integration/test/util/spark/SparkTableInfo.java

```diff
@@ -0,0 +1,143 @@
+/*
+ *  Copyright 2024 Datastrato Pvt Ltd.
+ *  This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.integration.test.util.spark;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import lombok.Data;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.Assertions;
+
+/** The output of SparkSQL is parsed to SparkTableInfo to check the result in test. */
+@Data
+public class SparkTableInfo {
+  private String name;
+  private String database;
+  private String comment;
+  private List<SparkColumnInfo> columns = new ArrayList<>();
+  private Map<String, String> tableProperties;
+  private List<String> unknownItems = new ArrayList<>();
+
+  public SparkTableInfo() {}
+
+  public String getTableName() {
+    return name;
+  }
+
+  // Include database name and table name
+  public String getTableIdentifier() {
+    if (StringUtils.isNotBlank(database)) {
+      return String.join(".", database, name);
+    } else {
+      return name;
+    }
+  }
+
+  @Data
+  public static class SparkColumnInfo {
+    private String name;
+    private String type;
+    private String comment;
+
+    private SparkColumnInfo(String name, String type, String comment) {
+      this.name = name;
+      this.type = type;
+      this.comment = comment;
+    }
+
+    public static SparkColumnInfo of(String name, String type) {
+      return of(name, type, null);
+    }
+
+    public static SparkColumnInfo of(String name, String type, String comment) {
+      return new SparkColumnInfo(name, type, comment);
+    }
+  }
+
+  public enum ParseStage {
+    COLLUMN,
+    DETECT,
+    PARTITION,
+    PARTITIONBUCKET,
+    TABLE_INFO,
+  }
+
+  public static SparkTableInfo getSparkTableInfo(List<Object[]> rows) {
+    ParseStage stage = ParseStage.COLLUMN;
+    SparkTableInfo tableInfo = new SparkTableInfo();
+    for (Object[] os : rows) {
```

Comment:
Could use  org.apache.spark.sql.connector.catalog.Table.properties and partitioning?
If you use Hive, maybe you can use Hive client to get table information, too.

Reply:
I prefer to parse spark sql result:
1. it's general and could reused by other catalogs like Iceberg and JDBC
2. it's necessary to check Spark SQL output

Reply:
It has bad compatible issues. If Spark community change the output, it will bring issues to you.

Reply:
yes, seems hard to get spark table meta